src/CMakeLists.txt

```diff
@@ -51,6 +53,56 @@ target_link_libraries(
 add_library(mp-units::mp-units ALIAS mp-units)
 install(TARGETS mp-units EXPORT mp-unitsTargets)
 
+# C++ modules
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
```

Comment:
I would prefer to have a CMake option for C++ modules so we can test both modes (headers and modules) with the same CMake installed on a local or CI machine.

Reply:
Of course, CMake could produce an error when MODULES opinion is selected, and the version is lower than required.

Reply:
It already backfired, see: https://github.com/mpusz/mp-units/actions/runs/7263359786/job/19788472479?pr=350#step:7:267.